File: lang/java/src/main/java/ru/dz/mqtt_udp/Engine.java
package ru.dz.mqtt_udp;

import ru.dz.mqtt_udp.util.GenericPacket;
import ru.dz.mqtt_udp.util.Throttle;
import ru.dz.mqtt_udp.util.mqtt_udp_defs;

public class Engine {

	private static boolean signatureRequired = false;
	private static String signatureKey = "signPassword";

	
	public static boolean isSignatureRequired() {		return signatureRequired;	}
	public static void setSignatureRequired(boolean req) {		signatureRequired = req;	}
	
	
	public static String getSignatureKey() {		return signatureKey;	}
	public static void setSignatureKey(String key) {		signatureKey = key;	}

	
	private static Throttle t = new Throttle();
	private static int maxReplyQoS = 1; // By default - TODO - add to defs!
	//private static int maxReplyQoS = 3; // for test


	/**
	 * Set packet send rate.
	 * @param msec average time in milliseconds between packets. Set to 0 to turn throttling off.
	 */
	public static void setThrottle(int msec) {		t.setThrottle(msec);	}
	
	/**
	 * Must be called in packet send code.
	 * Will put caller asleep to make sure packets are sent in a right pace. 
	 */
	public static void throttle() {				t.throttle();			}

	public static String getVersionString() {
		return String.format("%d.%d", mqtt_udp_defs.PACKAGE_VERSION_MAJOR, mqtt_udp_defs.PACKAGE_VERSION_MINOR);
	}
	public static int getMaxReplyQoS() { return maxReplyQoS; }
	public static void setMaxReplyQoS(int newMax) { maxReplyQoS = newMax; }
	
	
	
	public static void queueForResend(GenericPacket genericPacket) {
		
		
	}

}

File: lang/kotlin/mqtt_udp/src/main/kotlin/dz/ru/mqtt_udp/kt/Engine.kt
package dz.ru.mqtt_udp.kt;

import dz.ru.mqtt_udp.kt.io.UdpIO

public class Engine()
{
    private var signatureRequired : Boolean = false;
    private var signatureKey : String = "signPassword"; // TODO empty?

    public fun isSignatureRequired(): Boolean {        return signatureRequired    }
    public fun setSignatureRequired(req: Boolean) {        signatureRequired = req    }
    public fun getSignatureKey(): String {        return signatureKey    }
    public fun setSignatureKey(key: String) {        signatureKey = key    }


    private val t = Throttle()
    private val io = UdpIO()

    private val r = PacketReceiver(io);
    fun start() {
        r.requestStart()
    }

    fun setThrottle( msec : Int ) { t.setThrottle(msec); }


    private var maxReplyQoS = 1 // By default - TODO - add to defs!

    public fun getMaxReplyQoS(): Int {        return maxReplyQoS    }
    public fun setMaxReplyQoS(newMax: Int) {        maxReplyQoS = newMax    }


    companion object {
        }

}